sound/soc/sof/intel/hda-bus.c

```diff
@@ -74,14 +74,34 @@ static const struct hdac_io_ops io_ops = {
 	.dma_free_pages = sof_hda_dma_free_pages,
 };
 
+static u64 bus_idx_bits;
+
+static int get_free_index(void)
+{
+	int i = 0;
+
+	while ((bus_idx_bits >> i) & 0x1) {
+		if (i++ == 63)
```

Comment:
this code looks a bit ugly, aren't there macros to make things less weird?

Reply:
yep, that's why I add [RFC], I didn't feel good about it, only wanted to align if we need do it like this.